src/transformers/utils/hub.py

```diff
@@ -221,7 +221,8 @@ def extract_commit_hash(resolved_file: Optional[str], commit_hash: Optional[str]
     """
     if resolved_file is None or commit_hash is not None:
         return commit_hash
-
+    if isinstance(resolved_file, str) and sys.platform == "win32":
+        resolved_file = re.sub("\\\\", "/", resolved_file)
```

Comment:
I think the conversion can be done all the time. It would be more readable to use `r"\\"` instead of four backslashes.

Or maybe we could leverage `Path.as_posix` here to have the replacement done when necessary.

```suggestion
    resolved_file = str(Path(resolved_file.as_posix()))
```

Reply:
I'd also be in favor of the `as_posix()` solution.

Reply:
Okay, I pushed a commit to switch to using Pathlib's built in facility for this, and added a call to .resolve() as an additional bit of safety, as precedent or subsequent calls to resolve the file might reintroduce the double backslashes, had that happen before. 💢

Here's a quick sanity check on the behavior:
![image](https://user-images.githubusercontent.com/2738686/193114970-1f92ebbc-998e-4e39-9921-bc02ec8abc76.png)